### DISCUSSION TITLE
Whitelist sender email address on rate limit

### DISCUSSION THREAD

**Discussion by mohammad-hachem** [rspamd/rspamd#4034] [General]
In the rate limit module of Rspamd you can whitelist a recipient, IP address or an authenticated user.
But there is no option to whitelist a sender email.
This is needed in case there is a specific no-reply or broadcast email address that send a high rate of legitimate emails from time to time and we don't want our rate limit rule to be applied on its messages.

**Comment by vstakhov** [comment]
You can easily [add condition](https://rspamd.com/doc/lua/rspamd_config.html#m75087) to a symbol and use [maps in Lua](https://rspamd.com/doc/tutorials/writing_rules.html#using-maps-from-lua-plugin).

**Comment by mohammad-hachem** [comment]
Symbol and maps allow me to add a certain score when a message is received from a specific email address, but this will not prevent the rate limit module from (soft rejecting) the message since the rate limit will take place regardless of the score once the bucket is full.
